src/eos-autoupdater.c

```diff
@@ -105,6 +106,86 @@ get_stamp_dir (void)
                         UPDATE_STAMP_DIR);
 }
 
+static void
+log_with_msgid (const gchar *msgid,
+                GLogLevelFlags log_level,
+                const gchar *numeric_log_level,
+                const gchar *format,
+                va_list args)
+{
+  g_autofree gchar *message = NULL;
+  /* Apparently the version of GCC in Endless ignores the
+   * G_GNUC_PRINTF annotation that has a zero as the second parameter,
+   * so it suggests to use this attribute. */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsuggest-attribute=format"
+  gint message_length = g_vasprintf (&message, format, args);
+#pragma GCC diagnostic pop
+  const GLogField fields[] = {
+    { "MESSAGE", message, message_length },
+    { "MESSAGE_ID", msgid, EOS_UPDATER_MSGID_LENGTH },
+    { "PRIORITY", numeric_log_level, 1 },
```

Comment:
I had a look here - https://git.gnome.org/browse/glib/tree/glib/gmessages.c#n1402

It tells something about the priority being only 1 byte only with a following nul byte, but that sounds bogus - the link to the syslog manpage states no such restriction.

Will fix it to `-1`.